### ISSUE TITLE
[Deeplab v3 +] When I train in my own dataset, the result_segmentation is not exact. What 's happen

### ISSUE THREAD

**Issue by urgonguyen** [tensorflow/models#4284]
My data is Food17:18 class(17 object + 1 background)
+ train: 1370
+ val: 1226
+ trainval: 2596
I changed some thing in code:

+ segmentation_dataset.py
`_FOOD_SEG_INFORMATION = DatasetDescriptor(
    splits_to_sizes={
        'train':1370,
        'trainval':2596,
        'val': 1226,
    },
    num_classes=18,
    ignore_label=255,
)`

+train.py
`flags.DEFINE_boolean('initialize_last_layer', False,
                     'Initialize the last layer.')
+ get_dataset_colormap.py
_DATASET_MAX_ENTRIES = {
    _CITYSCAPES: 19,
    _PASCAL: 256,
    _FOOD: 18,
}`

`def create_food_label_colormap():
  """Creates a label colormap used in PASCAL VOC segmentation benchmark.

  Returns:
    A Colormap for visualizing segmentation results.
  """
  colormap = np.zeros((_DATASET_MAX_ENTRIES[_FOOD], 3), dtype=int)
  ind = np.arange(_DATASET_MAX_ENTRIES[_FOOD], dtype=int)

  for shift in reversed(range(8)):
    for channel in range(3):
      colormap[:, channel] |= bit_get(ind, channel) << shift
    ind >>= 3

  return colormap`

And the result is: 
https://drive.google.com/file/d/1ePj4I_FeoLzlVosFJ7RKiIbZkzMbtxu2/view?usp=sharing

Help me!

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
What is the top-level directory of the model you are using
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by kushagraagrawal** [comment]
Hi, what were the training parameters that you used? and the mIoU achieved?

**Comment by urgonguyen** [comment]
I use all of parameter from pascal voc 2012 in project. And I need to train again in my own data. I run train.py but have problem with path so I copy one file from local_test.sh and deleted all of lines under this:
`NUM_ITERATIONS=10
python "${WORK_DIR}"/train.py \
  --logtostderr \
  --train_split="trainval" \
  --model_variant="xception_65" \
  --atrous_rates=6 \
  --atrous_rates=12 \
  --atrous_rates=18 \
  --output_stride=16 \
  --decoder_output_stride=4 \
  --train_crop_size=513 \
  --train_crop_size=513 \
  --train_batch_size=4 \
  --training_number_of_steps="${NUM_ITERATIONS}" \
  --fine_tune_batch_norm=true \
  --tf_initial_checkpoint="${INIT_FOLDER}/deeplabv3_pascal_train_aug/model.ckpt" \
  --train_logdir="${TRAIN_LOGDIR}" \
--dataset_dir="${PASCAL_DATASET}"`
 I change NUM_ITERATIONS=30000 , --train_batch_size=1, --train_split="train"
and train on my dataset. But the result is not good( like the picture in my issue)

**Comment by kushagraagrawal** [comment]
It probably happened because of the batch size.

**Comment by urgonguyen** [comment]
Thank you and sorry I reply late, You mean I should increase train_batch_size.

**Comment by kushagraagrawal** [comment]
Yeah, you could keep a large batch size, which might help.

**Comment by kayvonkhosrowpour** [comment]
@urgonguyen Did you end up getting this to work? I'm trying to use my own custom dataset as you are, but I am having issues with a custom number of classes. Could you provide how you train the network and visualize inferences?
